master/buildbot/process/buildstep.py

```diff
@@ -831,6 +831,18 @@ def _connectPendingLogObservers(self):
                 self._pendingLogObservers.remove((logname, observer))
 
     def addURL(self, name, url):
+        """
+        Adds the given URL as a hyperlink in the build step's Artifacts
+        URLs must start with 'http://' or they will be treated as relative to the current build's URL
+        @param name: the displayed text for the hyperlink
+        @param url: the URL to add.
+        """
+
+        # Replace keywords in the URL with actual build-exclusive variables
+        if "<<BuilderName>>" in url:
+            url = url.replace("<<BuilderName>>", "%s" % self.build.builder.name)
+        if "<<BuildNumber>>" in url:
+            url = url.replace("<<BuildNumber>>", "%s" % self.build.build_status.number)
```

Comment:
Ah, I didn't realise format() could work that way. That does seem more ideal then.